doc/threading_model.md

```diff
@@ -0,0 +1,31 @@
+﻿# gRPC Threading Model
+
+## Introduction
+This document serves as a basic intro for the gRPC threading model. gRPC generates service code in two fashions - synchronous and asynchronous API. These two closely correlate to the threading model in play. 
+
+## Synchronous API Threading model
+This is the easiest model to start with and could be a satisfactory model for a large number of services. As such, the implementation is efficient (and is not a ‘toy’ implementation). In this model, gRPC takes care of the threading completely and the application code need to worry only about handling of the actual rpc call.
+* When the server starts up, a thread pool is created to handle incoming rpc requests. The default implementation of this thread pool is called DynamicThreadPool. It creates threads equal to number of processors in the system and all of them are in waiting state.
+* DynamicThreadPool has in­built scaling mechanism to create additional threads on demand. For integrators interested in having a little more control on the behavior of the thread pool, they can define GRPC_CUSTOM_DEFAULT_THREAD_POOL and provide the implementation of the thread pool interface (which is literally just 1 function that adds work to the pool).

```

Comment:
So the last two paragraphs are outdated since the the grpc++ server has been using ThreadManager since v1.1, not DynamicThreadPool (DynamicThreadPool is only used now for security credential management, not for actual RPCs). The operation of ThreadManager is similar but more efficient. I think it's ok to include these paragraphs for now on the grounds that this is the way at works at 1.0 , but we should immediately followup with a 2-paragraph fix explaining ThreadManager instead. Suggest comments from @sreecha